src/wallet/wallet.cpp

```diff
@@ -245,7 +245,7 @@ bool CWallet::Unlock(const SecureString& strWalletPassphrase, bool anonymizeOnly
 {
     SecureString strWalletPassphraseFinal;
 
-    if (!IsLocked())
+    if (!IsLocked() && !fWalletUnlockAnonymizeOnly) // was already fully unlocked, not only for anonymization
```

Comment:
So why not just change the flag when going from "unlocked for anonymization" to "fully unlocked" ?